Doc edits to batching, codegen, and some reorg (#3584)
apollographql/apollo-android: docs/source/advanced/response-based-codegen.mdx

}
```

## The `responseBased` codegen

The `responseBased` codegen differs from the `operationBased` codegen in the following ways:

* Generated models have a **1:1 mapping** with the JSON structure received in an operation's response.

> * Polymorphism is handled by generating **interfaces**. Possible shapes are then defined as different classes that implement a particular interface.

Reviewer: A class can implement multiple interfaces:

```suggestion
* Polymorphism is handled by generating **interfaces**. Possible shapes are then defined as different classes that implement the required interfaces.
```

  }
}
```

#### Accessors

> As a convenience, the `responseBased` codegen generates methods with the name pattern `as<Typename>` (e.g., `asDroid` or `asHuman`) that enable you to avoid manual casting:

Reviewer: It's not always "Typename". For an example: 

```graphql
query GetAnimal {
  ... on Pet {
    name
  }
  ... on WarmBlooded {
    temperature
  }
}
```

That will have `asPetWarmBlooded()` to avoid generating classes for each and every concrete type. We're [trying to define proper terminology for this](https://apollographql.quip.com/d7KaAOA9f8bz/Client-Glossary) which is not easy. I'd suggest "Shape" for the time being:

```suggestion
As a convenience, the `responseBased` codegen generates methods with the name pattern `as<ShapeName>` (e.g., `asDroid` or `asHuman`) that enable you to avoid manual casting:
```

}
```

#### Accessors

> As a convenience, the `responseBased` codegen generates methods with the name pattern `<typename>Fields` (e.g., `droidFields` or `humanFields`) that enable you to avoid manual casting.

Reviewer: ```suggestion
As a convenience, the `responseBased` codegen generates methods with the name pattern `<fragmentName>` (e.g., `droidFields` here or `humanDetails` if you named your fragment "HumanDetails") that enable you to chain calls.
```
